cmd/sops/main.go

```
@@ -332,3 +348,164 @@ func rotate(c *cli.Context, file string, fileBytes []byte, output io.Writer) err
+	tree.Branch = branch
+	ks, err := getKeysources(c, file)
+	if err != nil {
+		return tree, err
+	}
+	tree.Metadata.UnencryptedSuffix = c.String("unencrypted-suffix")
+	tree.Metadata.Version = "2.0.0"
+	tree.Metadata.KeySources = ks
+	key, err := tree.GenerateDataKey()
+	if err != nil {
+		return tree, cli.NewExitError(err.Error(), exitCouldNotRetrieveKey)
+	}
+	tree.Metadata.UpdateMasterKeys(key)
+	return tree, nil
+}
+
+func edit(c *cli.Context, file string, fileBytes []byte) error {
+	var tree sops.Tree
+	var stash map[string][]interface{}
+	var err error
+	if fileBytes == nil {
+		tree, err = loadExample(c, file)
+	} else {
+		tree, stash, err = decryptFile(store(file), fileBytes, c.Bool("ignore-mac"))
+	}
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not load file: %s", err), exitCouldNotReadInputFile)
+	}
+	tmpdir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not create temporary directory: %s", err), exitCouldNotWriteOutputFile)
+	}
+	defer os.RemoveAll(tmpdir)
+	tmpfile, err := os.Create(path.Join(tmpdir, path.Base(file)))
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not create temporary file: %s", err), exitCouldNotWriteOutputFile)
+	}
+	var out []byte
+	if c.Bool("show-master-keys") {
+		out, err = store(file).MarshalWithMetadata(tree.Branch, tree.Metadata)
+	} else {
+		out, err = store(file).Marshal(tree.Branch)
+	}
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not marshal tree: %s", err), exitErrorDumpingTree)
+	}
+	_, err = tmpfile.Write(out)
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not write output file: %s", err), exitCouldNotWriteOutputFile)
+	}
+	origHash, err := hashFile(tmpfile.Name())
+	if err != nil {
+		return cli.NewExitError(fmt.Sprintf("Could not hash file: %s", err), exitCouldNotReadInputFile)
+	}
+	invalidSyntax := true
+	for invalidSyntax || tree.Metadata.MasterKeyCount() == 0 {
```

It looks like invalidSyntax is only false at the very bottom. This variable can be replaced with just `break` at the bottom of the loop. If MasterKeyCount == 0, should this throw some sort of error to the client? It looks like this silently skips over editing the file.